R/commonly_used_db_data.R

```diff
@@ -53,14 +53,16 @@ get_touchstone_id <- function(con, touchstone) {
 ##' @param external_population_estimates The rationales are 1. we can use external population estimates if any and if necessary;
 ##' 2. demographic uncertainty not only affects models, but also FVPs. If we are to conduct sensitivity analysis on impact_by_year_of_vaccination, we need to vary population input for adjusting FVPs.
 ##' @param full_description TRUE if including scenario_descriptions (coverage estimates will be duplicated for scenarios); and FALSE if only providing coverage estimates
+##' @param demographic_source this is a 2nd option for getting population data from montagu
```

Comment:
I think the doc string for this arg should include info from `get_population` i.e. include some info about what this value actually needs to be. I think probably also worth putting a note of the two different modes of running into the function details after the title.

Probably also worth adding some validation at the start of this function which checks that only one of `touchstone_pop` or `demographic_source` is set and not both or neither. Returning a useful error message.

Reply:
The two parameters are hierachical.
  ## when touchstone_pop is null, touchstone_cov is used for touchstone_pop
  ## when demographic_source is not null, touchstone_pop is not used but ruled out
I have added a few lines of codes to explain and validate.